### ISSUE TITLE
fix trade_no field

### ISSUE THREAD

**Issue by dcsan** [dcsan/OpenMpTools#14]
```js
out_trade_no: `orderid_${moment().second()}${moment().millisecond()}`,
```

this needs to be a number that changes with every order.

    var milliseconds = (new Date).getTime();

can milliseconds?

**Comment by yangleyi** [comment]
No, for example, when there are many users, there will be two users placing orders at the same time at a certain time, then there will be a failure to place orders.

（不行，比如当有很多用户的时候，刚好在某一时刻有两个用户同时下单，那么就会有一个下单失败。）

**Comment by yangleyi** [comment]
**So, in order to determine the uniqueness of the order, I think there may be several solutions:**
1.  in order to generate, can be in the database to retrieve the order number，if it repeats will generate an order number again，until don't repeat. The disadvantage is that if there are already many orders, it will waste some time.
2. Make the order number as long as possible, and add a random number after the timestamp in the front, so as not to repeat it as much as possible.

_所以为了确定订单的唯一性，我认为可以有几种解决方式：_
1. 在生成订单的时候，可以在数据库检索一下订单号，如果它重复的话就再次生成一个订单号，直到不重复为止。缺点是如果已经有很多订单的话，这样会浪费一些时间；
2. 让订单号尽可能的长，前边为时间戳后边再添加一段随机数，为了尽可能的不重复

**Comment by dcsan** [comment]
or we could do `user.openid + timeStamp`
so each user cannot place multiple orders in the same millisecond.

**Comment by dcsan** [comment]
or
https://www.npmjs.com/package/uuid

**Comment by yangleyi** [comment]
Well yes.
